### PR TITLE
feat: match reference algorithmically

### LINKED ISSUE #57: match verse references algorithmically
## acceptance criteria
* Use an algorithm to fuzzy match book names
* Generate the regex to match verse references from the format string
* Remove the book match arrays from the translation strings
* Add bible namespace for es

### PR DESCRIPTION
**Pull request by Pertempto**
Closes #57

### CODE REVIEW

File: packages/web/src/public/locales/es/bible.json
Comment (Pertempto) [inline packages/web/src/public/locales/es/bible.json:1]: From https://catholic-resources.org/Bible/Abbreviations-Abreviaciones.htm

File: packages/web/src/features/translation/verse-utils.ts [hunk 1]
```diff
@@ -132,38 +133,56 @@ export function bookName(bookId: number, t: TFunction) {
   return t(bookKeys[bookId - 1], { ns: 'bible' });
 }
 
+/**
+ * Convert a reference format string for a language to a reference matching regex for that language.
+ * @param formatString The reference format string. See i18n.md for docs.
+ * @returns A regex for matching references.
+ */
+export function convertFormatToRegex(formatString: string): RegExp {
+  // TODO: improve this to increase language support.
```
Comment (Pertempto) [inline packages/web/src/features/translation/verse-utils.ts:142]: I left this for the future, if we find languages that need additional things beyond what this conversion supports.

File: packages/web/src/features/translation/VerseSelector.tsx [hunk 2]
```diff
@@ -20,7 +20,7 @@ export function VerseSelector({ verseId, onVerseChange }: VerseSelectorProps) {
   const verseInfo = parseVerseId(verseId);
 
   const onKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
-    if (e.key == 'Enter') {
+    if (e.key === 'Enter') {
```
Comment (Pertempto) [inline packages/web/src/features/translation/VerseSelector.tsx:23]: Addressed a lint warning.

File: package.json [hunk 3]
```diff
@@ -9,6 +9,7 @@
   "private": true,
   "dependencies": {
     "@prisma/client": "4.11.0",
+    "fuzzysort": "2.0.4",
```
Comment (Pertempto) [inline package.json:12]: This library gives us a sorted list of matches for a query string. Right now, I just the top match. In the future, we could use this list of matches to populate an auto-complete-style suggestion dropdown.

File: packages/web/src/features/translation/verse-utils.ts [hunk 4]
```diff
@@ -132,38 +133,56 @@ export function bookName(bookId: number, t: TFunction) {
...
  * @returns The verse ID if it can be determined, otherwise `null`.
  */
 export function parseReference(reference: string, t: TFunction): string | null {
-  // Parse the reference into three parts.
-  const referenceRegex = new RegExp(
-    t('reference_regex', { ns: 'bible' }) as string
+  const referenceRegex = convertFormatToRegex(
+    t('reference_format', { ns: 'bible' }) as string
   );
+  console.log('REGEX:', referenceRegex);
```
Comment (arrocke) [inline packages/web/src/features/translation/verse-utils.ts]:
```suggestion
```

**Review by arrocke — APPROVED** [review]
This works super well! Nice work

**Pull request merged**